### PR TITLE
Category Header implementation

### PR DESCRIPTION
**Pull request by marchi-martius**

### CODE REVIEW

File: app/components/CategoryOverviewView.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React, { Children, PropTypes, cloneElement } from 'react';
+import { View } from 'react-native';
+
+/*
+ *Passing category to all children as prop
+ */
+const CategoryOverviewView = ({ category, children }) => (
+  <View>
+    {Children.map(children, child => cloneElement(child, { category }))}
```
Comment (obliviusm) [inline app/components/CategoryOverviewView.js]: looks quite complicated and implicit. I hadn't known where `CategoryHeader` gets `category` from until i found this piece.

**Review by obliviusm — CHANGES_REQUESTED** [review]
also i got this warning in tests:
```
Warning: Failed prop type: The prop `category` is marked as required in `CategoryHeader`, but its value is `undefined`.
          in CategoryHeader (created by Main)
          in Main (created by FreemarketRn)
          in Provider (created by App)
          in App (created by FreemarketRn)
          in FreemarketRn
```
could you fix that?

File: app/components/CategoryOverviewView.js (hunk 1 above)
Comment (obliviusm) [reply app/components/CategoryOverviewView.js]: can you explain me why we need to do it this way and not with simple `<CategoryHeader {category=category}/>`?

**Comment by obliviusm** [comment]
BTW header looks cool

File: app/components/CategoryOverviewView.js (hunk 1 above)
Comment (marchi-martius) [reply app/components/CategoryOverviewView.js]: My intent was to make platform-independent component which can be used to retrieve current category (and products etc) and pass it as props to platform-specific components (header, list, footer). 
I don't particularly like my solution either. I'll rework it simpler.

**Comment by marchi-martius** [comment]
@obliviusm fixed, please check out.

**Review by obliviusm — APPROVED** [review]

**Comment by obliviusm** [comment]
Cool. Looks better to me
1. I reverted react-native upgrade. hope you don't mind. reason: sometimes i get blank screen when run project in this branch. i believe upgrade is the problem.
2. FYI i'm working on this right now: https://github.com/marchi-martius/FreemarketRn/issues/21

**Pull request merged**